### PR TITLE
MAINT: Fix misc. typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell`

### CODE REVIEW

File: numpy/core/src/npymath/halffloat.c [hunk 1]
```diff
@@ -281,7 +281,7 @@ npy_uint16 npy_floatbits_to_halfbits(npy_uint32 f)
     if (f_exp <= 0x38000000u) {
         /*
          * Signed zeros, subnormal floats, and floats with small
-         * exponents all convert to signed zero halfs.
```
Comment (rkern) [inline numpy/core/src/npymath/halffloat.c]: This is an intentional spelling, I think.

File: doc/changelog/1.13.0-changelog.rst [hunk 2]
```diff
@@ -364,7 +364,7 @@ A total of 309 pull requests were merged for this release.
 * `#8928 <https://github.com/numpy/numpy/pull/8928>`__: BUG: runtests --bench fails on windows
 * `#8929 <https://github.com/numpy/numpy/pull/8929>`__: BENCH: Masked array benchmarks
 * `#8939 <https://github.com/numpy/numpy/pull/8939>`__: DEP: Deprecate `np.ma.MaskedArray.mini`
-* `#8942 <https://github.com/numpy/numpy/pull/8942>`__: DOC: stop refering to 'S' dtype as string
```
Comment (rkern) [inline doc/changelog/1.13.0-changelog.rst]: We may want to leave the changelogs alone to show exactly what the title of the PRs are, misspellings and all.

Comment (eric-wieser) [reply doc/changelog/1.13.0-changelog.rst]: Or we can rename the PRs

File: numpy/core/src/npymath/halffloat.c (hunk 1 above)
Comment (charris) [reply numpy/core/src/npymath/halffloat.c]: `half-floats` is what they are.

File: numpy/core/src/npymath/halffloat.c [hunk 3]
```diff
@@ -396,7 +396,7 @@ npy_uint16 npy_doublebits_to_halfbits(npy_uint64 d)
     if (d_exp <= 0x3f00000000000000ULL) {
         /*
          * Signed zeros, subnormal floats, and floats with small
-         * exponents all convert to signed zero halfs.
+         * exponents all convert to signed zero halves.
```
Comment (charris) [inline numpy/core/src/npymath/halffloat.c]: `half-floats`.

**Comment by charris** [comment]
Corrected, and looks like the PRs have also been corrected. Thanks @luzpaz .

**Pull request merged**